netlink-packet-route/src/rtnl/link/nlas/link_infos.rs

```diff
@@ -714,6 +724,137 @@ impl<'a, T: AsRef<[u8]> + ?Sized> Parseable<NlaBuffer<&'a T>> for InfoVxlan {
+                | EncapSport(ref value)
+                | EncapDport(ref value)
+            => NativeEndian::write_u16(buffer, *value),
+            Link(ref value)
+                | FwMark(ref value)
+            => NativeEndian::write_u32(buffer, *value),
+            Local(ref value)
+                | Remote(ref value)
+            => buffer.copy_from_slice(value.as_slice()),
+        }
+    }
+
+    fn kind(&self) -> u16 {
+        use self::InfoIpTun::*;
+
+        match self {
+            Unspec(_) => IFLA_IPTUN_UNSPEC,
+            Link(_) => IFLA_IPTUN_LINK,
+            Local(_) => IFLA_IPTUN_LOCAL,
+            Remote(_) => IFLA_IPTUN_REMOTE,
+            Ttl(_) => IFLA_IPTUN_TTL,
+            Tos(_) => IFLA_IPTUN_TOS,
+            Proto(_) => IFLA_IPTUN_PROTO,
+            Pmtudisc(_) => IFLA_IPTUN_PMTUDISC,
+            EncapType(_) => IFLA_IPTUN_ENCAP_TYPE,
+            EncapFlags(_) => IFLA_IPTUN_ENCAP_FLAGS,
+            EncapSport(_) => IFLA_IPTUN_ENCAP_SPORT,
+            EncapDport(_) => IFLA_IPTUN_ENCAP_DPORT,
+            CollectMetadata(_) => IFLA_IPTUN_COLLECT_METADATA,
+            FwMark(_) => IFLA_IPTUN_FWMARK,
+        }
+    }
+}
+
+impl<'a, T: AsRef<[u8]> + ?Sized> Parseable<NlaBuffer<&'a T>> for InfoIpTun {
+    fn parse(buf: &NlaBuffer<&'a T>) -> Result<Self, DecodeError> {
+        use self::InfoIpTun::*;
+        let payload = buf.value();
+        //println!("{} - {:?}", buf.kind(), payload);
+        Ok(match buf.kind() {
+            IFLA_IPTUN_UNSPEC => Unspec(payload.to_vec()),
+            IFLA_IPTUN_LINK => Link(parse_u32(payload).context("invalid IFLA_IPTUN_LINK value")?),
+            IFLA_IPTUN_LOCAL => Local(payload.to_vec()),
+            IFLA_IPTUN_REMOTE => Remote(payload.to_vec()),
+            IFLA_IPTUN_TTL => Ttl(parse_u8(payload).context("invalid IFLA_IPTUN_TTL value")?),
+            IFLA_IPTUN_TOS => Tos(parse_u8(payload).context("invalid IFLA_IPTUN_TOS value")?),
+            IFLA_IPTUN_PROTO => Proto(parse_u8(payload).context("invalid IFLA_IPTUN_PROTO value")?),
+            IFLA_IPTUN_PMTUDISC => {
+                Pmtudisc(parse_u8(payload).context("invalid IFLA_IPTUN_PMTUDISC value")?)
+            }
+            IFLA_IPTUN_ENCAP_TYPE => {
+                EncapType(parse_u16(payload).context("invalid IFLA_IPTUN_ENCAP_TYPE value")?)
+            }
+            IFLA_IPTUN_ENCAP_FLAGS => {
+                EncapFlags(parse_u16(payload).context("invalid IFLA_IPTUN_ENCAP_FLAGS value")?)
+            }
+            IFLA_IPTUN_ENCAP_SPORT => {
+                EncapSport(parse_u16_be(payload).context("invalid IFLA_IPTUN_ENCAP_SPORT value")?)
+            }
+            IFLA_IPTUN_ENCAP_DPORT => {
+                EncapDport(parse_u16_be(payload).context("invalid IFLA_IPTUN_ENCAP_DPORT value")?)
+            }
+            IFLA_IPTUN_COLLECT_METADATA => CollectMetadata(
+                parse_u8(payload).context("invalid IFLA_IPTUN_COLLECT_METADATA value")?,
```

Comment:
The kernel code indicate, once `IFLA_IPTUN_COLLECT_METADATA` exist in netlink message, it means `CollectMeta(true)`.